pytket/phir/sharding/sharder.py

```diff
@@ -273,11 +272,8 @@ def should_op_create_shard(op: Op) -> bool:
             `True` if the operation is one that should result in shard creation
         """
         return (
-            op.type in (SHARD_TRIGGER_OP_TYPES)
-            or (
-                op.type == OpType.Conditional
-                and cast(Conditional, op).op.type in (SHARD_TRIGGER_OP_TYPES)
-            )
+            op.type in SHARD_TRIGGER_OP_TYPES
+            or isinstance(op, Conditional)
```

Comment:
I can't remember why I had the `op.op.type` check in here, but it seems like this is more correct.